### DISCUSSION TITLE
how to prefetch data before the template render using $fetch from server api.

### DISCUSSION THREAD

**Discussion by shahensaifullah** [nuxt/nuxt#16733] [Questions] [answered]
if client send to request to server and server response and keep that response the client as "data".
Sometime, when i try to see first array of data[0], it shows 0 is not available but if i make condition is data.length, code is execute well.
but i want to call server and set to a variable before the template render.
in nuxt2, there was fetch(){} method to call api which is executed before the template is render. but in nuxt3 using setup script how can i do that?

**Comment by fayazara** [comment, marked as answer]
https://v3.nuxtjs.org/api/composables/use-async-data/
